sustaingym/envs/cogen/env.py

```diff
@@ -154,13 +155,15 @@ def _forecast_from_time(self, day: int, time_step: int) -> pd.DataFrame:
         for the following self.forecast_horizon + 1 time steps."""
         slice_df = self.ambients_dfs[day].iloc[time_step:min(time_step+self.forecast_horizon+1, self.timesteps_per_day)]
         # fix so that if the slice_df is not long enough, it will take the first values of the next day
-        # TODO: figure out what to do if we're on the last day and there is no next day
         if len(slice_df) < self.forecast_horizon + 1:
             slice_df = pd.concat([slice_df, self.ambients_dfs[day+1].iloc[:self.forecast_horizon + 1 - len(slice_df)]])
         cols = ['Ambient Temperature', 'Ambient Pressure',
                 'Ambient rel. Humidity', 'Target Net Power',
                 'Target Process Steam', 'Energy Price', 'Gas Price']
-        return slice_df[cols].astype(np.float32)
+        forecast = slice_df[cols].astype(np.float32)
+        # add iid gaussian noise to future observations
+        forecast.iloc[1:] += self.forecast_noise_std*np.random.randn(self.forecast_horizon, 7)
```

Comment:
Please use `self.np_random.normal` (see the [Gymnasium API](https://gymnasium.farama.org/api/env/#gymnasium.Env.np_random) and the numpy [Generator.normal()](https://numpy.org/doc/stable/reference/random/generated/numpy.random.Generator.normal.html) documentation) instead of `np.random.randn`. We need to use `self.np_random` in order to respect the random seed that is passed into `env.reset()`.